src/main/webapp/app/components/tabs/curation-data-import-tab/import-methods.tsx

```diff
@@ -0,0 +1,286 @@
+import { AuthStore } from 'app/stores';
+
+export type GeneDI = {
+  hugo_symbol: string;
+};
+
+export type GenericDI = GeneDI & {
+  value: string;
+};
+
+export type GenomicIndicatorDI = GeneDI & {
+  genomic_indicator: string;
+  allele_state?: string;
+  description?: string;
+};
+
+export type MutationDI = GeneDI & {
+  alteration: string;
+  name?: string;
+  protein_change?: string;
+};
+
+export type SomaticMutationDI = MutationDI & {
+  oncogenicity: string;
+  mutation_effect?: string;
+  description?: string;
+};
+
+export type GermlineMutationDI = MutationDI & {
+  pathogenicity: string;
+  description?: string;
+};
+
+const alleleStateCheck = async (
+  alleleStates: string[],
+  onValidAlleleStates: () => Promise<DataImportStatus>,
+): Promise<DataImportStatus> => {
+  let status = new DataImportStatus();
+
+  const unmappedAlleleStates: string[] = [];
+  if (alleleStates) {
+    // validate allele state is correct
+    alleleStates.forEach(alleleState => {
+      if (!Object.values(ALLELE_STATE).includes(alleleState as ALLELE_STATE)) {
+        unmappedAlleleStates.push(alleleState);
+      }
+    });
+  }
+  if (unmappedAlleleStates.length > 0) {
+    status.status = 'error';
+    status.message = `${unmappedAlleleStates.join(', ')} ${pluralize('is', unmappedAlleleStates.length)} not supported. Only ${Object.values(ALLELE_STATE).join(', ')} are supported.`;
+  } else {
+    status = await onValidAlleleStates();
+  }
+  return Promise.resolve(status);
+};
+
+const impactStatusCheck = (status: string, allowedStatus: string[]): DataImportStatus => {
+  const importStatus = new DataImportStatus();
+
+  if (status) {
+    if (!allowedStatus.includes(status)) {
+      importStatus.status = 'error';
+      importStatus.message = `${status} not supported. Only ${allowedStatus.join(', ')} are supported.`;
+    }
+  }
+  return importStatus;
+};
+
+const oncogenicityCheck = (pathogenicity: string): DataImportStatus => {
+  return impactStatusCheck(pathogenicity, uniq(Object.values(FIREBASE_ONCOGENICITY)));
+};
+
+const pathogenicityCheck = (pathogenicity: string): DataImportStatus => {
+  return impactStatusCheck(pathogenicity, Object.values(PATHOGENICITY));
+};
+
+export const geneCheck = async (
+  firebaseGeneService: FirebaseGeneService,
+  isGermline: boolean,
+  hugoSymbol: string,
+  onGeneExists: () => Promise<DataImportStatus>,
+) => {
+  const status = new DataImportStatus();
+  hugoSymbol = hugoSymbol.trim();
+  if (hugoSymbol) {
+    const genePath = getFirebaseGenePath(isGermline, hugoSymbol);
+    const geneData = await firebaseGeneService.getObject(genePath);
+    if (geneData.exists()) {
+      return onGeneExists();
+    } else {
+      status.status = 'error';
+      status.message = 'Gene does not exist. Please create before importing';
+    }
+  } else {
+    status.status = 'error';
+    status.message = 'Gene is missing from data';
+  }
+  return Promise.resolve(status);
```

Comment:
I don't think you have to wrap this in a promise since the function is already asynchronous.